### PR TITLE
Updating pip and pip-tools

### PR DESCRIPTION
**Pull request by kronos30**
Updating PIP and PIP-Tools to a newer version.

### CODE REVIEW

**Review by paulbovbel — APPROVED** [review]

**Comment by paulbovbel** [comment]
CI tests failing on focal - we might need to condition this on the platform OS

**Review by paulbovbel — CHANGES_REQUESTED** [review]
CI tests failing on focal - we might need to condition this on the platform OS or the python version

**Comment by kronos30** [comment]
> CI tests failing on focal - we might need to condition this on the platform OS or the python version

Yeah.. that makes sense.

**Comment by paulbovbel** [comment]
builtin_venv is a helpful hint that it's python3, so can just move this into the conditional below

File: catkin_virtualenv/src/catkin_virtualenv/venv.py [hunk 1]
```diff
@@ -65,8 +65,8 @@ def initialize(self, python, use_system_packages, extra_pip_args, clean=True):
             raise RuntimeError(error_msg)
 
         preinstall = [
-            "pip==20.1",
-            "pip-tools==5.1.2",
+            "pip==21.3.1",
```
Comment (paulbovbel) [inline catkin_virtualenv/src/catkin_virtualenv/venv.py:68]: does this support python2 and python3, or just the latter?

if we're going to drop python2 support entirely, probably worth updating the README

Comment (kronos30) [reply catkin_virtualenv/src/catkin_virtualenv/venv.py:68]: This version just support python3.6 to 3.10. Yep will update the read-me

**Pull request closed without merging**